editor/js/libs/app.js

```diff
@@ -1,37 +1,40 @@
+import { ViewportPathtracer } from '../Viewport.Pathtracer.js';
+
 var APP = {
 
 	Player: function () {
```

Comment:
Have you considered doing:

```js
Player: function ( renderer = null ) {

    if ( renderer === null ) renderer = new THREE.WebGLRenderer( { antialias: true } );

}
```

That way we can do `new Player( new PathTracingRenderer() )` and we should be able to keep the current untouched?

Reply:
Do you mean that using single `Player` to handle both `playing app` and `rendering video`?

Reply:
Trying to look for ways to implement this while modifying the current code as less are possible.